### PR TITLE
ci: setup handler to clean up old states

### PR DESCRIPTION
**Pull request by cosmic-vagabond**
### What has Changed?

* remove old CW codes (binaries) to save space
* remove AH CW contract states entirely as we migrated all AH functionalities to Tier native module

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/elys-network/elys/blob/main/docs/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/elys-network/elys/blob/main/CONTRIBUTING.md#testing)
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### Deployment Notes

Are there any specific considerations to take into account when deploying these changes? This may include new dependencies, scripts that need to be executed, or any aspects that can only be evaluated in a deployed environment.

### Screenshots and Videos

Please provide any relevant before and after screenshots by uploading them here. Additionally, demo videos can be highly beneficial in demonstrating the process.

### CODE REVIEW

File: app/setup_handlers.go [hunk 1]
```diff
@@ -22,6 +24,37 @@ func setUpgradeHandler(app *ElysApp) {
...
+				// Retrieve the wasm module store key
+				storeKey := app.keys[wasmmodule.StoreKey]
+
+				// Retrieve the wasm module store
+				store := ctx.KVStore(storeKey)
+
+				// List of prefixes to clear
+				prefixes := [][]byte{
+					// Account History contract
+					wasmtypes.GetContractStorePrefix(sdk.MustAccAddressFromBech32("elys17p9rzwnnfxcjp32un9ug7yhhzgtkhvl9jfksztgw5uh69wac2pgs98tvuy")),
```
Comment (amityadav0) [inline app/setup_handlers.go:37]:
```suggestion
					wasmtypes.GetContractStorePrefix(sdk.MustAccAddressFromBech32("elys1s37xz7tzrru2cpl96juu9lfqrsd4jh73j9slyv440q5vttx2uyesetjpne")),
```

Comment (amityadav0) [reply app/setup_handlers.go:37]: I think this is the AH address

**Review by amityadav0 — APPROVED** [review]

File: app/setup_handlers.go [hunk 2]
```diff
@@ -22,6 +24,37 @@ func setUpgradeHandler(app *ElysApp) {
...
+				}
+
+				// Add old code keys to the list of prefixes to clear
+				for i := uint64(1); i < 671; i++ {
```
Comment (amityadav0) [inline app/setup_handlers.go:41]: are we sure this won't affect the TS functionality ? just confirming

Comment (amityadav0) [reply app/setup_handlers.go:41]: what if old contracts make a call to these IDs ? will that halt the chain with invalid mem error ?

File: app/setup_handlers.go (hunk 1 above)
Comment (cosmic-vagabond) [reply app/setup_handlers.go:37]: good catch thanks @amityadav0

File: app/setup_handlers.go (hunk 2 above)
Comment (cosmic-vagabond) [reply app/setup_handlers.go:41]: @amityadav0 a cosmwasm code is the binary of the contract stored on the chain, it is not the actual instance/state of the contract, when we deploy a new version of the contract, we first store the new code of the contract binary (that creates a new code id) and then we migrate the contract from the previous code id to the new one, so any past code ids are unused. AH, TS and FS are using the code IDs 671, 672 and 673 so we are safe.

Comment (amityadav0) [reply app/setup_handlers.go:41]: got it, makes sense

**Pull request merged**